test/unit/UFragments.ts

```diff
@@ -1,15 +1,53 @@
 import { ethers, upgrades } from 'hardhat'
-import { Contract, Signer, BigNumber } from 'ethers'
+import { Contract, Signer, BigNumber, BigNumberish } from 'ethers'
 import { expect } from 'chai'
+import { fromRpcSig } from 'ethereumjs-util'
+import { TypedDataUtils, signTypedMessage, TypedMessage } from 'eth-sig-util'
```

Comment:
Could use ethers for typed data signing: https://docs.ethers.io/v5/api/signer/#Signer-signTypedData

Reply:
Done 👍